catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -223,6 +250,12 @@ private void fetchNextResults() {
       }
     }
 
+    private boolean isDistinctResult(Result result) {
+      return result.getMetacard() == null
+          || result.getMetacard().getId() == null
+          || !foundIds.contains(result.getMetacard().getId());
+    }
```

Comment:
If null metacards/ids were considered to be not distinct, could the stream filters on lines 234 and 236 be removed? Line 222 would have already eliminated them from the result set.

Reply:
@lessarderic What conditions, if any, can result in a result with a null metacard or a metacard with a null id?

Reply:
I think Ethan put those checks in because he couldn't be sure they wouldn't be null.